tempesta_fw/t/functional/helpers/control.py

```diff
@@ -320,23 +395,28 @@ def letout_finish(self):
         if self.module_stap:
             self.node.run_cmd('rm -r %s' % self.modules_dir)
 
-    def start(self):
-        Tempesta.start(self)
+    def run_start(self):
+        Tempesta.run_start(self)
         self.inject_prepare()
         self.inject()
 
-    def stop(self):
-        self.letout()
-        self.letout_finish()
-        Tempesta.stop(self)
+    def stop_tempesta(self):
+        letout_fail = None
+        try:
+            self.letout()
```

Comment:
These to functions (`self.letout(); self.letout_finish()`) should be moved to `self.stop_procedures[]`. It will do just what you want.

Reply:
fixed